x-b2/src/Auth.js

```diff
@@ -0,0 +1,106 @@
+import Parse from 'parse/dist/parse.min.js';
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { Redirect, Route } from "react-router-dom";
+
+const Auth = {
+  isAuthenticated: false,
+  signin(cb) {
+    Auth.isAuthenticated = true;
+    if (cb) cb();  // Asegurarse de que cb no sea undefined antes de llamarlo
+  },
+  signout(cb) {
+    Auth.isAuthenticated = false;
+    if (cb) cb();  // Asegurarse de que cb no sea undefined antes de llamarlo
+  }
+};
+
+
+const authContext = createContext();
+
+export function ProvideAuth({ children }) {
+  const auth = useProvideAuth();
+
+  return (
+    <authContext.Provider value={auth}>
+      {children}
+    </authContext.Provider>
+  );
+}
+
+export function useAuth() {
+  return useContext(authContext);
+}
+
+export function useProvideAuth() {
+  const [user, setUser] = useState(null);
+
+  useEffect(() => {
+    // Comprobar si hay un usuario autenticado al cargar el componente
+    const fetchUser = async () => {
+      const currentUser = Parse.User.current();
+      if (currentUser) {
+        setUser(currentUser);
+      }
+    };
+    fetchUser();
+  }, []);
+
+  const signin = (cb) => {
+    console.log("Attempting to sign in"); // Log para depuración
+    Auth.signin(() => {
+      setUser(Parse.User.current());
+      console.log("User set:", Parse.User.current()); // Log para depuración
+      if (cb) cb();
+    });
+  };
+
+  const signout = (cb) => {
```

Comment:
Please implement the `Parse.User.Logout()` before the `setUser(null)`. Here is an example:
https://www.back4app.com/docs/react-native/parse-sdk/working-with-users/react-native-login#BpiHW
![Screenshot 2024-05-31 at 1 04 47 PM](https://github.com/Albertwo512/X-B2/assets/20372706/33eb6a38-d880-4c94-8608-5fb9a3102b25)